angel_system/impls/detect_activities/two_stage/two_stage_detect_activities.py

```diff
@@ -0,0 +1,186 @@
+                if torch.cuda.is_available():
+                    model_device = torch.device(device=self._torch_device)
+                    model = model.to(device=model_device)
+                else:
+                    raise RuntimeError(
+                        "Use of CUDA requested but not available."
+                    )
+            else:
+                model_device = torch.device(self._torch_device)
+
+            self._model = model
+            self._model_device = model_device
+
+        return model
+
+    def detect_activities(
+        self,
+        frame_iter: Iterable[np.ndarray],
+        aux_data_iter: Iterable[Dict]
+    ) -> Iterable[str]:
+        """
+        Formats the given iterable of frames into the required input format
+        for the swin model and then inputs them to the model for inferencing.
+        """
+        # Check that we got the right number of frames
+        frame_iter = list(frame_iter)
+        aux_data = dict(aux_data_iter)
+        assert all([len(frame_iter) == len(aux_data[i]) for i in aux_data])
+        # assert len(frame_iter) == (self._sampling_rate * self._num_frames)
+        model = self.get_model()
+
+        # Apply data pre-processing
+        frames = [self.transform(f) for f in frame_iter]
+        frames = torch.stack(frames)
+        for k in aux_data:
+            aux_data[k] = torch.Tensor(aux_data[k])
+
+        # Move the inputs to the GPU if necessary
+        if self._model.cuda:
+            frames = frames.cuda()
+            for k in aux_data:
+                aux_data[k] = aux_data[k].cuda()
+
+        # Predict!
+        with torch.no_grad():
+            preds = self._model(frames, aux_data)
+
+        # Get the top predicted classes
+        post_act = torch.nn.Softmax(dim=1)
+        preds: torch.Tensor = post_act(preds) # shape: (1, num_classes)
+        top_preds = preds.topk(k=5)
+
+        # Map the predicted classes to the label names
+        # top_preds.indices is a 1xk tensor
+        pred_class_indices = top_preds.indices[0]
+
+        pred_class_names = [self._labels[int(i)] for i in pred_class_indices]
+
+        # Filter out any detections below the threshold
+        predictions = []
+        pred_values = top_preds.values[0]
+        for idx, p in enumerate(pred_class_names):
+            if (pred_values[idx] > self._det_threshold):
+                predictions.append(p)
+
+        # pdb.set_trace()
+        return predictions
+
+    def get_config(self) -> dict:
+        return {
+            "cuda_device": self._cuda_device,
+            "num_classes": self._num_classes,
+            "det_threshold": self._det_threshold,
+            "checkpoint_path": self._checkpoint_path,
+            "num_frames": self._num_frames,
+            "sampling_rate": self._sampling_rate,
+            "labels_file": self._labels_file,
+        }
+
+    @classmethod
+    def is_usable(cls) -> bool:
+        # Only torch/torchvision required
+        return True
```

Comment:
This does not need to be implemented here as the super-class implementation already just returns true.

Reply:
Ok. Removed.